tests/unit/s2n_error_lookup_test.c

```diff
@@ -86,6 +94,18 @@ int main(void)
     EXPECT_EQUAL(strcmp(s2n_strerror_name((S2N_ERR_T_USAGE + 1) << S2N_ERR_NUM_VALUE_BITS), "Internal s2n error"), 0);
     EXPECT_EQUAL(strcmp(s2n_strerror((S2N_ERR_T_USAGE + 1) << S2N_ERR_NUM_VALUE_BITS, "EN"), "Internal s2n error"), 0);
 
+    /* Ensure the file/line information is returned as expected */
+    s2n_result_ignore(test_function(false));
+    EXPECT_EQUAL(strcmp(s2n_strerror_source(S2N_ERR_SAFETY), "s2n_error_lookup_test.c:25"), 0);
+
+/* constructs a debug string from a path */
+#define EXAMPLE_DEBUG_STR(path) (_S2N_EXTRACT_BASENAME(_S2N_DEBUG_LINE_PREFIX path))
+
```

Comment:
I'm really not a fan of mid-function #defines and think they're a slippery slope, but if you really like this one it's not obviously affecting readability.

Reply:
would you prefer it at the top?